esp-hal/build.rs

```diff
@@ -180,6 +180,26 @@ fn main() -> Result<(), Box<dyn Error>> {
     copy_dir_all(&config_symbols, "ld/sections", &out)?;
     copy_dir_all(&config_symbols, format!("ld/{device_name}"), &out)?;
 
+    // supply build time and date
+    // see https://reproducible-builds.org/docs/source-date-epoch/
+    let ts = match std::env::var("SOURCE_DATE_EPOCH") {
```

Comment:
this is obviously not the time when some source files got touched for the last time but when the build script was run

I _guess_ that is fine since production builds are usually clean builds